lib/fast_jsonapi/serialization_core.rb

```diff
@@ -99,21 +99,42 @@ def to_json(payload)
       end
 
       # includes handler
-
       def get_included_records(record, includes_list, known_included_objects)
-        includes_list.each_with_object([]) do |item, included_records|
-          object_method_name = @relationships_to_serialize[item][:object_method_name]
-          record_type = @relationships_to_serialize[item][:record_type]
-          serializer = @relationships_to_serialize[item][:serializer].to_s.constantize
-          relationship_type = @relationships_to_serialize[item][:relationship_type]
-          included_objects = record.send(object_method_name)
-          next if included_objects.blank?
-          included_objects = [included_objects] unless relationship_type == :has_many
-          included_objects.each do |inc_obj|
-            code = "#{record_type}_#{inc_obj.id}"
-            next if known_included_objects.key?(code)
-            known_included_objects[code] = inc_obj
-            included_records << serializer.record_hash(inc_obj)
+        return unless includes_list.present?
+
+        includes_list.sort.each_with_object([]) do |include_item, included_records|
+          items = include_item.to_s.include?('.') ? include_item.to_s.split('.').map{|item| item.to_sym} : [include_item]
+          remaining_items = nil
+          if items.size > 1
+            items_copy = items.dup
+            items_copy.delete_at(0)
+            remaining_items = [items_copy.join('.').to_sym]
+          end
+          item_to_serialize = items.last
+
+          items.each do |item|
+            next unless relationships_to_serialize[item]
```

Comment:
https://github.com/LifeTales/fast_jsonapi/blob/49e7e86e93c61fa5cf34278144378c8640d9f079/lib/fast_jsonapi/object_serializer.rb#L154-L155

`relationships_to_serialize` might be `nil`, throwing `undefined method `[]' for nil:NilClass`; need to catch that too.

Reply:
made corresponding PR for this at https://github.com/LifeTales/fast_jsonapi/pull/1